src/e2ee/RustEngine.ts

```diff
@@ -106,6 +107,14 @@ export class RustEngine {
         settings.rotationPeriod = BigInt(encEv.rotationPeriodMs);
         settings.rotationPeriodMessages = BigInt(encEv.rotationPeriodMessages);
 
+        await this.run(RequestType.KeysQuery);
```

Comment:
This doesn't appear to cause a query request to happen?

Reply:
It does, because by the time this is hit, the CryptoClient has updated the list of tracked users in the room, which queues outgoing key query requests in the crypto state machine.

Without this line, the first message a bot sends in an encrypted room with a user it hasn't seen before will be unencryptable by that user.

Otherwise, the SDK calls `RustEngine.run` (and processes all outgoing requests, including key queries) _after_ having already sent its first (encrypted) message, meaning it won't have the device key of new users & can't send them the to-device message containing the room key.